os/kernel/binary_manager/binary_manager_verify.c

```diff
@@ -55,10 +55,11 @@ static int binary_manager_verify_header_data(int type, void *header_input)
 	if (type == BINARY_KERNEL) {
 		kernel_binary_header_t *header_data = (kernel_binary_header_t *)header_input;
 		if (header_data->header_size == 0 || header_data->binary_size == 0 \
-		|| header_data->version < KERNEL_BIN_VER_MIN || header_data->version > KERNEL_BIN_VER_MAX) {
+		|| header_data->version < KERNEL_BIN_VER_MIN || header_data->version > KERNEL_BIN_VER_MAX \
+		|| header_data->header_size != sizeof(kernel_binary_header_t) - CHECKSUM_SIZE) {
```

Comment:
sizeof(kernel_binary_header_t) - CHECKSUM_SIZE can be 0?
Or you can remove first condition, 'header_data->header_size == 0'

Reply:
Modified.